File: thrift/src/main/csharp/Query/Proxy/Client/QueryClient.cs
using System;
using System.Collections.Generic;

using Thrift.Transport;
using Thrift.Protocol;
using Query.Proxy.Thrift.Protocol;

namespace CSharpQueryClient
{
    class QueryClient
    {
        static void Main(string[] args)
        {
            string host = "server.query-proxy.local";
            TTransport transport = new TSocket(host, 9966);
            TProtocol protocol = new TBinaryProtocol(transport, true, true);
            transport.Open();
            QueryProxyService.Client client = new QueryProxyService.Client(protocol);

            QueryParams queryParams = new QueryParams();
            queryParams.Table = "collection1";
            queryParams.Type = QueryType.SOLR;
            queryParams.ParamList = new List<string>();
            queryParams.ParamList.Add("q=上海");
            queryParams.ParamList.Add("fl=*");
            queryParams.ParamList.Add("fq=building_type:1");
            queryParams.ParamList.Add("start=50");
            queryParams.ParamList.Add("rows=10");
            queryParams.ParamList.Add("wt=json");

            QueryResult result = client.query(queryParams);
            Console.WriteLine("offset=" + result.Offset);
            Console.WriteLine("length=" + result.Length);
            foreach (string record in result.Results)
            {
                Console.WriteLine("record=" + record);
            }
            
            Console.Read();
        }
    }
}

File: thrift/src/main/python/queryproxy/client/QueryClient.py
#!/usr/bin/python

# coding=utf-8
from thrift.transport import TSocket
from thrift.transport import TTransport
from thrift.protocol import TBinaryProtocol
from queryproxy.thrift.protocol.QueryProxyService import Client

def query():
    try:
        transport = TSocket.TSocket('server.query-proxy.local', 9966)
        transport = TTransport.TBufferedTransport(transport)
        protocol = TBinaryProtocol.TBinaryProtocol(transport)
        transport.open()
        
        Client client = new Client(protocol)
        client.query()
    except TException, tx:
        print '%s' % (tx.message) 


if __name__ == '__main__':
    query()